src/dotnet/ToolPackage/ToolPackageObtainer.cs

```diff
@@ -95,16 +100,49 @@ IProjectRestorer projectRestorer
                 packageVersion = concreteVersion;
             }
 
-            ToolConfiguration toolConfiguration = GetConfiguration(packageId: packageId, packageVersion: packageVersion, individualToolVersion: toolDirectory);
+            LockFile lockFile = new LockFileFormat()
+                .ReadWithLock(toolDirectory.WithFile("project.assets.json").Value)
+                .Result;
 
-            return new ToolConfigurationAndExecutableDirectory(
+            LockFileItem dotnetToolSettings = FindAssetInLockFile(lockFile, "DotnetToolSettings.xml", packageId);
+
+            if (dotnetToolSettings == null)
+            {
+                throw new PackageObtainException("Cannot find DotnetToolSettings");
```

Comment:
Will it be possible to install a tool with a missing settings file (despite a package type check)?  If so, I think this message should be localized and a little bit friendlier, something along the lines of `Package '{packageId}' is missing tool settings file DotNetToolSettings.xml.`

Reply:
> Will it be possible to install a tool with a missing settings file (despite a package type check)?

Yes, NuGet will not check the setting file.

And yes, I am missing loc. And your error message is better